chain/client/src/sync/state/network.rs

```diff
@@ -69,18 +69,21 @@ impl StateSyncDownloadSourcePeerSharedState {
                 None => PartIdOrHeader::Header,
             },
         };
-        if let Some(request) = self.pending_requests.get(&key) {
-            if request
-                .peer_id
-                .as_ref()
-                .is_some_and(|expecting_peer_id| expecting_peer_id == &peer_id)
```

Comment:
The heart of this change is essentially replacing this `is_some_and` with `is_none_or`. However, is_none_or is still unstable in rust version 1.81.0 so I had to reorganize things a bit.